pkg/cli/mirror/catalog_images.go

```diff
@@ -229,12 +228,11 @@ func (o *MirrorOptions) buildCatalogLayer(ctx context.Context, srcRef, targetRef
 
 	logrus.Debugf("Pulling image %s for processing", srcRef)
 	// Pull source reference image
-	ref, err := name.ParseReference(srcRef)
+	ref, err := name.ParseReference(srcRef, o.getNameOpts()...)
 	if err != nil {
 		return err
 	}
-	options := o.getRemoteOpts(ctx)
-	img, err := remote.Image(ref, options...)
+	img, err := remote.Image(ref, o.getRemoteOpts(ctx)...)
```

Comment:
Is there a reason to re-create these options on each call?

Reply:
When I was initially writing this section, I stored the options in the MirrorOptions struct, but it broke the unit test since the `remote.Option` and `name.Option` types are functions that take pointer arguments. I created a helper function as an alternative, but I am totally open to suggestions on better ways to do this.